lib/build/source-inclusion.js

```diff
@@ -47,26 +47,28 @@ exports.SourceInclusion = class {
   }
 
   addAllMatchingResources(loaderConfig) {
-    const vfs = require('vinyl-fs');
+    return new Promise(resolve => {
+      const vfs = require('vinyl-fs');
 
-    let bundler = this.bundle.bundler;
-    let root = path.resolve(bundler.project.paths.root, loaderConfig.path);
-    let pattern = path.resolve(bundler.project.paths.root, this.pattern);
+      let bundler = this.bundle.bundler;
+      let root = path.resolve(bundler.project.paths.root, loaderConfig.path);
+      let pattern = path.resolve(bundler.project.paths.root, this.pattern);
 
-    var subsume = (file, cb) => {
-      let filePath = file.path;
-      let moduleId = path.join(loaderConfig.name, filePath.replace(root, ''));
-      moduleId = moduleId.replace(/\\/g, '/');
-      let ext = path.extname(moduleId);
-      moduleId = moduleId.substring(0, moduleId.length - ext.length);
+      var subsume = (file, cb) => {
+        let filePath = file.path;
+        let moduleId = path.join(loaderConfig.name, filePath.replace(root, ''));
+        moduleId = moduleId.replace(/\\/g, '/');
+        let ext = path.extname(moduleId);
+        moduleId = moduleId.substring(0, moduleId.length - ext.length);
 
-      let bundledSource = bundler.addFile(file, this);
-      bundledSource.moduleId = moduleId;
+        let bundledSource = bundler.addFile(file, this);
+        bundledSource.moduleId = moduleId;
 
-      cb(null, file);
-    };
+        cb(null, file);
+      };
 
-    vfs.src(pattern).pipe(mapStream(subsume));
+      vfs.src(pattern).pipe(mapStream(subsume)).on('end', resolve);
```

Comment:
Should there be any error catching here in case the tracing fails? Just want to make sure that it not only _works_, but that when it _fails_ the user knows what happened (i.e. should there also be a `reject` catch in there?)